mesonbuild/build.py

```diff
@@ -2091,6 +2092,15 @@ def __init__(
             compilers: T.Dict[str, 'Compiler'],
             kwargs):
         self.prelink = T.cast('bool', kwargs.get('prelink', False))
+        if StaticLibrary.allow_thin_archives and 'cuda' in environment.coredata.compilers[for_machine]:
+            # nvcc chokes on thin archives:
+            #   nvlink fatal   : Could not open input file 'libfoo.a.p'
+            #   nvlink fatal   : elfLink internal error
+            # hence we disable them if 'cuda' is enabled globally. See also
+            # - https://github.com/mesonbuild/meson/pull/9453
+            # - https://github.com/mesonbuild/meson/issues/9479#issuecomment-953485040
```

Comment:
Should we mark this commit as closing that PR?

Reply:
I mean, this commit just makes "cuda work" but dodges the overall question of users being able to disable thin archives. I think the usecase in that PR is a bit extreme (i.e. trying to manually invoke the compiler etc.) but that's not my call to make.

Reply:
Yeah, fair enough. Maybe a case for "Bug:" instead of "Closes:"?

Reply:
done